packages/headless/src/controllers/facets/category-facet/headless-category-facet.ts

```diff
@@ -179,23 +301,14 @@ export function buildCategoryFacet(
       const canShowLessValues = values.length > options.numberOfValues;
 
       return {
-        /** The facet id. */
         facetId,
-        /** The parent values of the facet. */
         parents,
-        /** The values of the facet. */
         values,
-        /** `true` if a search is in progress and `false` otherwise. */
         isLoading,
-        /** `true` if there is at least one non-idle value and `false` otherwise. */
         hasActiveValues,
-        /** `true` if there are more values to display and `false` otherwise. */
         canShowMoreValues,
-        /** `true` if fewer values can be displayed and `false` otherwise. */
         canShowLessValues,
-        /** The active sortCriterion of the facet. */
         sortCriteria: request!.sortCriteria,
-        /** The state of the facet's searchbox. */
         facetSearch: facetSearch.state,
       };
     },
```

Comment:
One thing I would strongly recommend before merging is checking out the `master` branch and writing code that uses all the properties on the category facet controller. The taking that, switching to this branch, and using the code as a test that the new interfaces exactly match the old with no breaking changes.

e.g.

```
const categoryFacet = buildCategoryFacet({} as any, {} as any);
categoryFacet.toggleSelect()
categoryFacet.deselectAll()
etc.
```

I would also do the same for category facet options interface.